web/modules/custom/joinup_communities/tallin/src/Plugin/Field/FieldWidget/TallinEntryWidget.php

```diff
@@ -0,0 +1,171 @@
+   *   The widget settings.
+   * @param array $third_party_settings
+   *   Any third party settings.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   */
+  public function __construct($plugin_id, $plugin_definition, FieldDefinitionInterface $field_definition, array $settings, array $third_party_settings, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($plugin_id, $plugin_definition, $field_definition, $settings, $third_party_settings);
+
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $plugin_id,
+      $plugin_definition,
+      $configuration['field_definition'],
+      $configuration['settings'],
+      $configuration['third_party_settings'],
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $item = $items[$delta];
+
+    $element['wrapper'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->fieldDefinition->getLabel(),
+    ];
+
+    $element['wrapper']['tallin_description'] = [
+      '#type' => 'html_tag',
+      '#tag' => 'h3',
+      '#value' => $this->fieldDefinition->getSetting('tallin_description'),
+      '#weight' => 0,
+    ];
+
+    $element['wrapper']['explanation'] = [
+      '#type' => 'text_format',
+      '#title' => t('Explanations'),
+      '#default_value' => $item->value,
+      '#format' => $item->format,
+      '#weight' => 1,
+    ];
+
+    $element['wrapper']['status'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Implementation status'),
+      '#options' => $this->getStatusOptions(),
+      '#default_value' => $item->status,
+      '#weight' => 2,
+    ];
+
+    $element['wrapper']['uri'] = [
+      '#type' => 'url',
+      '#title' => $this->t('Related website'),
+      '#default_value' => $item->uri,
+      '#maxlength' => 2048,
+      '#required' => $element['#required'],
+      // Only external links, i.e. full links.
+      '#link_type' => LinkItemInterface::LINK_EXTERNAL,
+      '#weight' => 3,
+    ];
+
+    $element['#element_validate'][] = [get_called_class(), 'validateFormElement'];
+    return $element;
+  }
+
+  /**
+   * Form element validation handler for the complete form element.
+   */
+  public static function validateFormElement($element, FormStateInterface $form_state, $form) {
+    $status = $element['wrapper']['status']['#value'];
+    $explanation = $element['wrapper']['explanation']['value']['#value'];
+    if ($status !== 'no_data' && empty($explanation)) {
+      $form_state->setError($element, t('An explanation is required. Please, fill in the <em>Explanations</em> field.'));
```

Comment:
`$this->t(...)`

Reply:
I tried to trigger this error. There is no reference to which field this is targetting, so it's hard to understand.